conjure_oxide/src/solvers/minion.rs

```diff
@@ -0,0 +1,279 @@
+//! Solver interface to minion_rs.
+
+use crate::ast::{
+    DecisionVariable, Domain as ConjureDomain, Expression as ConjureExpression,
+    Model as ConjureModel, Name as ConjureName, Range as ConjureRange,
+};
+use minion_rs::ast::{
+    Constant as MinionConstant, Constraint as MinionConstraint, Model as MinionModel,
+    Var as MinionVar, VarDomain as MinionDomain,
+};
+
+impl TryFrom<ConjureModel> for MinionModel {
+    // TODO: set this to equal ConjureError once it is merged.
+    type Error = String;
+
+    fn try_from(conjure_model: ConjureModel) -> Result<Self, Self::Error> {
+        let mut minion_model = MinionModel::new();
+
+        // We assume (for now) that the conjure model is fully valid
+        // i.e. type checked and the variables referenced all exist.
+        parse_vars(&conjure_model, &mut minion_model)?;
+        parse_exprs(&conjure_model, &mut minion_model)?;
+
+        Ok(minion_model)
+    }
+}
+
+fn parse_vars(conjure_model: &ConjureModel, minion_model: &mut MinionModel) -> Result<(), String> {
+    // TODO: remove unused vars?
+    // TODO: ensure all vars references are used.
+    for (name, variable) in conjure_model.variables.iter() {
+        parse_var(name, variable, minion_model)?;
+    }
+    Ok(())
+}
+
+fn parse_var(
+    name: &ConjureName,
+    variable: &DecisionVariable,
+    minion_model: &mut MinionModel,
+) -> Result<(), String> {
+    let str_name = name_to_string(name.to_owned());
+
+    // A Conjure-Oxide variable has an arbritary number of ranges as part of its domain. Here, we
+    // require that it has a single range in its domain, and that the domain is of integers.
+
+    let ranges = match &variable.domain {
+        ConjureDomain::IntDomain(range) => Ok(range),
+        x => Err(format!("Not supported: {:?}", x)),
+    }?;
+
+    if ranges.len() != 1 {
+        return Err(format!(
+            "Variable {:?} has {:?} ranges, expected 1.",
+            str_name,
+            ranges.len()
+        ));
+    }
+
+    let range = ranges
+        .first()
+        .ok_or(format!("Variable {:?} has no range", str_name))?;
+
+    // Because we have one range, it must be a bounded range, not a single point.
+    // Otherwise this would just be a constant.
+    // TODO correct?
```

Comment:
It's a reasonable assumption, though not required really. Minion would handle variables with a singleton domain just fine.